pac4j-cas/src/main/java/org/pac4j/cas/client/CasClient.java

```diff
@@ -202,20 +261,23 @@ protected CasCredentials retrieveCredentials(final WebContext context) throws Re
             final CasCredentials casCredentials = new CasCredentials(ticket, getName());
             logger.debug("casCredentials : {}", casCredentials);
             return casCredentials;
-        } else if (this.logoutHandler.isLogoutRequest(context)) {
+        }
+
+        if (this.logoutHandler.isLogoutRequest(context)) {
```

Comment:
Took out else clauses that were not needed. A return at the end of every block will terminate execution.